toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -107,14 +107,18 @@ func fetchPackages(dependencyGraph *pkggraph.PkgGraph, hasUnresolvedNodes, tryDo
 	}
 	defer cloner.Close()
 
+	toolchainPackages, err := schedulerutils.ReadReservedFilesList(*toolchainManifest)
+	if err != nil {
+		err = fmt.Errorf("unable to read toolchain manifest file '%s':\n%w", *toolchainManifest, err)
+		return
+	}
+	toolchainPackageSet := make(map[string]bool)
+	for _, pkg := range toolchainPackages {
+		toolchainPackageSet[pkg] = true
+	}
```

Comment:
Let's just make `resolveGraphNodes` accept a set instead. I think using a list there was a mistake to begin with and we'll simplify a bit of the code both here and there this way.

Reply:
After thinking about this more: since we want to restore the reverted toolkit changes and they are quite robust, I take that back - the refactoring can wait to minimize merge conflicts.